sensu/plugins/metric-os-api.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+#
+# Makes http calls to the different components to gather metric data
+#
+# return CRITICAL if any commands fail
+# return WARNING if no commands fail, but one or more timeout
+#
+#
+# Jose L Coello Enriquez <jlcoello@us.ibm.com>
+
+import argparse
+import re
+import sys
+import time
+import socket
+import os
+import json
+import requests
+
+STATE_OK = 0
+STATE_WARNING = 1
+STATE_CRITICAL = 2
+
+state = STATE_OK
+
+metric_data = ''
+
+DEFAULT_SCHEME = '{}'.format(socket.gethostname())
+
+
+def request_token(url, method='GET', retries=2, **kwargs):
+    r = None
+    try:
+        for i in range(retries + 1):
+            if i > 0:
+                time.sleep(2)
+            r = requests.request(method, url, **kwargs)
+            if r.status_code < 500:
+                break
+    except requests.exceptions.RequestException as e:
+        print "{0} returned {1}".format(url, e)
+        state = STATE_WARNING
+
+    return r
+
+def request(scheme, url, timestamp, method='GET', retries=2, **kwargs):
+    global state
```

Comment:
Removed all globals, added them to parameters of functions and returned tuples which where unpacked on main